incident/utils/category_field_values.py

```diff
@@ -0,0 +1,283 @@
+from incident.models import choices
+
+def basic_html_val(page, field):
+    # If no value for the attribute, return blank
+    if not getattr(page, field):
+        return ''
+
+    link = '{}?{}={}'.format(
+        page.get_parent().get_url(),
+        field,
+        getattr(page, field)
+    )
+
+    # If has a get_display function associated with the field, use that
+    # else use just the field value
+    if hasattr(page, 'get_{}_display'.format(field)):
+        value = getattr(page, 'get_{}_display'.format(field))().capitalize()
+    else:
+        value = getattr(page, field).capitalize()
+
+    return '<a href="{}" class="text-link">{}</a>'.format(
+        link,
+        value
+    )
+
+def boolean_html_val(page, field):
+    # If no value for the attribute, return blank
+    if not getattr(page, field):
+        return ''
+
+    link = '{}?{}={}'.format(
+        page.get_parent().get_url(),
+        field,
+        getattr(page, field)
+    )
+    value = 'Yes' if getattr(page, field) else 'No'
+
+    return '<a href="{}" class="text-link">{}</a>'.format(
+        link,
+        value
+    )
+
+def list_html_val(page, field):
+    # If no value for the attribute, return blank
+    if not len(getattr(page, field).all()):
+        return ''
+
+    val_list = []
+    for item in getattr(page, field).all():
+        link = '{}?{}={}'.format(
+            page.get_parent().get_url(),
+            field,
+            item.pk
+        )
+        value = item.title.capitalize()
+        val_list.append(
+            '<li class="details-table__list-item"><a href="{}" class="text-link">{}</a></li>'.format(
+            link,
+            value
+        ))
+
+    return '<ul class="details-table__list">{}</ul>'.format('\n'.join(val_list))
```

Comment:
A similar strategy could be employed on other functions here, particularly the ones that involve assembling lists.

Reply:
Definitely agree that it would be much cleaner. I did think about this. I was just thinking if a `render_to_string` would further affect the performance of our backend. But I don't know if there would be any substantial degradation. What do you think?